papers/scipy/scipy.rst

```diff
@@ -0,0 +1,1706 @@
+
+    Step response of the Butterworth bandpass filter with
+    order 10 and normalized cutoff frequencies 0.04 and 0.16.
+    :label:`fig-sos-bandpass-response-step`
+
+
+
+
+In the remaining examples of IIR filtering, we will use only the
+SOS representation.
+
+Lowpass filter
+--------------
+
+Figure :ref:`fig-pressure-example-input` shows a times series containing
+pressure measurements [SO]_. At some point in the interval 20 < t < 22,
+an event occurs in which the pressure jumps and begins oscillating
+around a "center".  The center of the oscillation decreases and
+appears to level off.
+
+.. figure:: figs/pressure_example_input.pdf
+
+   *Top*: Pressure, for the interval 15 < t < 35 (milliseconds).
+   *Bottom*: Spectrogram of the pressure time series (generated using a
+   window size of 1.6 milliseconds).
+   :label:`fig-pressure-example-input`
+
+We are not interested in the oscillations, but we are interested in the mean
+value around which the signal is oscillating.
+
+To preserve the slowly varying behavior while eliminating the high frequency
+oscillations, we'll apply a low-pass filter.  To apply the filter, we can
+use either ``sosfilt`` or ``sosfiltfilt`` from ``scipy.signal``.
+The function ``sosfiltfilt`` is a forward-backward filter--it applies the
+filter twice, once forward and once backward.  This effectively doubles the
+order of the filter, and results in zero phase shift.
+Because we are interesting in the "event" that occurs in 20 < t < 22,
+it is important to preserve the phase characteristics of the signal, so
+we use ``sosfiltfilt``.
+
+The following code snippet defines two convenience functions.  These
+functions allow us to specify the sampling frequency and the lowpass
+cutoff frequency in whatever units are convenient.  They take care of
+scaling the values to the units expected by ``scipy.signal.butter``.
+
+
+.. code-block:: python
+
+    from scipy.signal import butter, sosfiltfilt
+
+    def butter_lowpass(cutoff, fs, order):
+        normal_cutoff = cutoff / (0.5*fs)
+        sos = butter(order, normal_cutoff,
+                     btype='low', output='sos')
+        return sos
+
+    def butter_lowpass_filtfilt(data, cutoff, fs,
+                                order):
+        sos = butter_lowpass(cutoff, fs, order=order,
+                              output='sos')
+        y = sosfiltfilt(sos, data)
+        return y
+
+The results of filtering the data using ``sosfiltfilt`` are shown in
+Figure :ref:`fig-pressure-example-filtered`.
+
+.. figure:: figs/pressure_example_filtered.pdf
+
+   *Top*: Filtered pressure, for the interval 15 < t < 35 (milliseconds).
+   The light gray curve is the unfiltered data.
+   *Bottom*: Spectrogram of the filtered time series (generated using a
+   window size of 1.6 milliseconds).
+   The dashed line is at 1250 Hz.
+   :label:`fig-pressure-example-filtered`
+
+**Comments on creating a spectrogram.**
+The spectrograms in Figures :ref:`fig-pressure-example-input`
```

Comment:
Perhaps worth spending one sentence on what a spectrogram is? Also can mention ``signal.stft``, creates the same type of spectrogram.

Reply:
I added a sentence about what a spectrogram is.  I'm not sure about also mentioning `stft`, so I didn't.